test/ocamlorg_web/graphql_test.ml

```diff
@@ -0,0 +1,201 @@
+      { Package.Info.synopsis = "Decentralised installation system"
+      ; description = ""
+      ; authors = []
+      ; maintainers = []
+      ; license = "ISC"
+      ; homepage = []
+      ; tags = []
+      ; dependencies = []
+      ; depopts = []
+      ; conflicts = []
+      ; url = None
+      }
+  ; Package.create
+      ~name:(Package.Name.of_string "0install-gtk")
+      ~version:(Package.Version.of_string "2.1.0")
+      { Package.Info.synopsis = "Decentralised installation system - GTK UI"
+      ; description = ""
+      ; authors = []
+      ; maintainers = []
+      ; license = "ISC"
+      ; homepage = []
+      ; tags = []
+      ; dependencies = []
+      ; depopts = []
+      ; conflicts = []
+      ; url = None
+      }
+  ; Package.create
+      ~name:(Package.Name.of_string "abt")
+      ~version:(Package.Version.of_string "2.1.0")
+      { Package.Info.synopsis = "Package dependency solve"
+      ; description = ""
+      ; authors = []
+      ; maintainers = []
+      ; license = "ISC"
+      ; homepage = []
+      ; tags = []
+      ; dependencies = []
+      ; depopts = []
+      ; conflicts = []
+      ; url = None
+      }
+  ; Package.create
+      ~name:(Package.Name.of_string "ocaml")
+      ~version:(Package.Version.of_string "2.1.0")
+      { Package.Info.synopsis = "OCaml port of CMU's abstract binding trees"
+      ; description = ""
+      ; authors = []
+      ; maintainers = []
+      ; license = "ISC"
+      ; homepage = []
+      ; tags = []
+      ; dependencies = []
+      ; depopts = []
+      ; conflicts = []
+      ; url = None
+      }
+  ; Package.create
+      ~name:(Package.Name.of_string "absolute")
+      ~version:(Package.Version.of_string "2.1.0")
+      { Package.Info.synopsis =
+          "Interactive theorem prover based on lambda-tree syntax"
+      ; description = ""
+      ; authors = []
+      ; maintainers = []
+      ; license = "ISC"
+      ; homepage = []
+      ; tags = []
+      ; dependencies = []
+      ; depopts = []
+      ; conflicts = []
+      ; url = None
+      }
+  ]
+
+let start_with_test str1 str2 cond () =
+  Alcotest.(check bool)
+    "is true or false"
+    cond
+    (Ocamlorg_web.Graphql.starts_with str1 str2)
+
+let is_package_test name1 name2 cond () =
+  Alcotest.(check bool)
+    "is false"
+    cond
+    (Ocamlorg_web.Graphql.is_package name1 name2)
+
+let get_packages_result_test () =
+  let total_packages = List.length packages in
+  let offset = 0 in
+  let limit = 3 in
+  let filter = None in
+  let all_packages =
+    Ocamlorg_web.Graphql.get_packages_result
+      total_packages
+      offset
+      limit
+      filter
+      packages
+  in
+  let num_of_pakgs = List.length all_packages.packages in
+  Alcotest.(check int) "returns 3 packages" 3 num_of_pakgs
+
+let get_deps_test () =
+  let deps = Ocamlorg_web.Graphql.get_info dependencies in
+  let num_of_deps = List.length deps in
+  Alcotest.(check int) "returns 6 dependencies" 6 num_of_deps
```

Comment:
It is testing the the get_info function in graphql.ml file.
The goal was to test all small functions independently, hence the reason we have get_deps_test renamed to get_info_test, start_with_test, is_package_test and  get_single_package_test

Reply:
You seem to only test that `get_info` preserves the length of the list. Is that what you want to test?

Reply:
> You seem to only test that get_info preserves the length of the list. Is that what you want to test?
Nope, the test is to ensure that get_info works fine. I just used the length to pass it to Alcotest as length. I did not know how to compare the data so I used length here

Reply:
So far you've only used the most basic Alcotest testables like `int`, `string` and `bool`. However, it's also possible to construct more complex testables from those basic ones. For example, the return value of `get_info` is a list of a record called `package_info` containing a `string` and a `string option`. So you have two options here:
1. To construct a testable for the record `package_info`, you could do what [@patricoferris has explained in the context of `get_single_package_test`](https://github.com/ocaml/v3.ocaml.org-server/pull/73#discussion_r688377673): make it a custom testable by defining a pretty print function for it. Then you can use [Alcotest.list](https://mirage.github.io/alcotest/alcotest/Alcotest/index.html#val-list) to compare list of those records.
2. You can compare lists of pairs instead of comparing lists of `package_info` records. For that, you can simply also use the `Alcotest` testable constructors [Alcotest.pair](https://mirage.github.io/alcotest/alcotest/Alcotest/index.html#val-pair) and [Alcotest.option](https://mirage.github.io/alcotest/alcotest/Alcotest/index.html#val-list) additionally to [Alcotest.list](https://mirage.github.io/alcotest/alcotest/Alcotest/index.html#val-list) and write something like
```
Alcotest.(check (list (pair string (option string))))
```

Is that more or less understandable?